### DISCUSSION TITLE
scVelo on Seurat object

### DISCUSSION THREAD

**Discussion by vim15** [theislab/scvelo#458] [Q&A]
Hello, 

I've created a Seurat object consisting of 3 separate conditions ("A", "B", "C") and converted this object ("integrated") into a file compatible with scVelo as outlined in the Seurat vigniette.
I've loaded the object into scVelo and everything seems to be working fine in terms of the suggested pipeline to follow and the tutorials. 

As it's the integrated seurat that was used, is there a way to see if velocities or latent times (etc) change from 'stim' to 'stim' in the different conditions that are present? As to my understanding I am only seeing the overall picture. Any extra information you may have about dealing with integrated seurat objects in scVelo would be greatly appreciated.

**Comment by WeilerP** [comment]
@vim15 could you please elaborate on what you are trying to do? Do I understand correctly that your observations are grouped into the three clusters *A*, *B*, *C*? Now you wish to compare the inferred velocity and latent times between these clusters? What do you mean by *'stim' to 'stim'*?

**Reply by vim15** [reply]
Hi @WeilerP, 

No these are not 3 clusters. Consider you have 3 samples (e.g 3 conditions. WT, TreatmentA, TreatmentB). In Seurat when you integrate these objects, their clusters are anchored and you get 1 UMAP that way you can run diff expression between the same cluster across the different conditions, as (for example) your stem cell cluster is going to be expressing different genes in WT vs TreatmentA vs TreatmentB. (e.g https://satijalab.org/seurat/articles/integration_introduction.html). These treatments are generally saved in the Seurat object under 'stim', but you can call it whichever argument you prefer. 

When you convert this Seurat object into an object compatible with scVelo and load it, you are only seeing the main UMAP, and that is what is used to infer latent times, velocity etc. How can you segregate your conditions in such a way that you run velocity in each subsequent Treatment rather the object as a whole. 
I hope that makes more sense but if not please let me know.

**Reply by WeilerP** [reply]
Okay, gotcha - thanks for the explanation! ATM, we only support subsetting genes via `var_names` in `scv.tl.recover_dynamics`. To run e.g. `scv.tl.recover_dynamics` on different subsets of observations, you would have to subset the original AnnData and pass the new object to the function. This, however, would also require calculating a new neighbor graph and imputations:

```python
adata_subsetted = adata[obs_subset, :].copy()  # subset
sc.pp.moments(adata_subsetted)  # new PCA, kNN and moments
scv.tl.recover_dynamics(adata_subsetted)
```

**Reply by AAA-3** [reply]
Hello! 

I am exporting my Seurat derived cells which were integrated as explained above. My 'Stim' groups are WT cells and knock-out cells.

Based on you suggestion above, would I be correct in assuming that I should analyse RNA velocity on WT and KO cells separately (ie one WT-only AnnData  file and a second  KO-only AnnData file)? This is what I have been doing so far and I keep my original tSNE coordinates which I add into the AnnData files- if we are running new KNN and PCAs, should we also repeat the clustering in scanpy/scvelo or is it still ok to keep the tSNE coordinates?

As an aside, is there any plan to subset genes via 'stim'/'treatments' ?
